frontend/src/metabase/admin/settings/auth/components/GoogleAuthForm/GoogleAuthForm.tsx

```diff
@@ -58,67 +58,86 @@ const GoogleAuthForm = ({
-            }
-            placeholder={
-              isSsoEnabled
-                ? "mycompany.com, example.com.br, otherdomain.co.uk"
-                : "mycompany.com"
-            }
-            nullable
-            {...getFormFieldProps(settings[DOMAIN_KEY])}
+        <Form disabled={!dirty}>
+          <Breadcrumbs
+            crumbs={[
+              [t`Authentication`, "/admin/settings/authentication"],
+              [t`Google Sign-In`],
+            ]}
           />
-          <FormSubmitButton
-            title={isEnabled ? t`Save changes` : t`Save and enable`}
-            primary
-            disabled={!dirty}
-          />
-          <FormErrorMessage />
-        </GoogleForm>
+          <Stack gap="md" maw="32.5rem">
+            <Title order={2} mt="lg">{t`Sign in with Google`}</Title>
+            <Text c="text-medium">
+              {t`Allows users with existing Metabase accounts to login with a Google account that matches their email address in addition to their Metabase username and password.`}
+            </Text>
+            <Text c="text-medium">
+              {jt`To allow users to sign in with Google you'll need to give Metabase a Google Developers console application client ID. It only takes a few steps and instructions on how to create a key can be found ${(
+                <ExternalLink key="link" href={docsUrl}>
+                  {t`here`}
+                </ExternalLink>
+              )}.`}
+            </Text>
+            <FormTextInput
+              name={CLIENT_ID_KEY}
+              label={t`Client ID`}
+              placeholder={t`{your-client-id}.apps.googleusercontent.com`}
+              {...getEnvSettingProps(settingDetails?.[CLIENT_ID_KEY])}
+            />
+            <FormTextInput
+              name={DOMAIN_KEY}
+              label={t`Domain`}
+              description={
+                hasTokenFeature
+                  ? t`Allow users to sign up on their own if their Google account email address is from one of the domains you specify here:`
+                  : t`Allow users to sign up on their own if their Google account email address is from:`
+              }
+              placeholder={
+                hasTokenFeature
+                  ? "mycompany.com, example.com.br, otherdomain.co.uk"
+                  : "mycompany.com"
+              }
+              nullable
+              {...getEnvSettingProps(settingDetails?.[DOMAIN_KEY])}
+            />
+            <Box>
+              <FormSubmitButton
+                label={
+                  isGoogleAuthEnabled ? t`Save changes` : t`Save and enable`
+                }
+                variant="filled"
+                disabled={!dirty}
+              />
+            </Box>
+            <FormErrorMessage />
+          </Stack>
+        </Form>
       )}
     </FormProvider>
   );
 };
 
-const getFormFieldProps = (setting?: SettingDefinition) => {
+const getEnvSettingProps = (setting?: SettingDefinition) => {
```

Comment:
This is the first time I've seen us do env settings this way. I was just working on `SMTPConnectionForm.tsx` which uses wrapper component for the envs:

```
  <SetByEnvVarWrapper
    settingKey="email-smtp-username"
    settingDetails={settingsDetails?.["email-smtp-username"]}
  >
    <FormTextInput
    ...
     </SetByEnvVarWrapper>
```

It's not strictly an issue, but just highlighting that we're doing different things in different places.

Reply:
I like your component! spreading in readOnly props was the existing pattern that I just cleaned up a little